### PR TITLE
Remove all -1's as extremal elements. Replace members to optional

### PR DESCRIPTION
**Pull request by schaumb**

### CODE REVIEW

File: src/chart/options/channel.cpp [hunk 1]
```diff
@@ -89,10 +89,10 @@ bool Channel::isSeriesUsed(const Data::SeriesIndex &index) const
 	    || (dimensionIds.includes(index));
 }
 
-int Channel::findPos(const Data::SeriesIndex &index) const
+std::optional<int> Channel::findPos(const Data::SeriesIndex &index) const
```
Comment (simzer) [inline src/chart/options/channel.cpp:92]: findPos in Channel and Channels are not used, can be removed.

File: src/chart/ui/events.cpp [hunk 2]
```diff
@@ -30,7 +30,7 @@ std::string PointerEvent::dataToJson() const
 		coords = chart->getCoordSystem().getOriginal(position);
 	}
 	return "\"element\":\"" + elementUnder + "\""
-	     + ",\"pointerId\":" + std::to_string(pointerId)
+	     + ",\"pointerId\":" + std::to_string(pointerId.value_or(-1))
```
Comment (simzer) [inline src/chart/ui/events.cpp:33]: here would be better to put null to json instead of -1 in case of not set

**Review by simzer — APPROVED** [review]

File: src/chart/ui/events.cpp (hunk 2 above)
Comment (schaumb) [reply src/chart/ui/events.cpp]: I replaced to Conv::toString(pointerId).

File: src/chart/options/channel.cpp (hunk 1 above)
Comment (schaumb) [reply src/chart/options/channel.cpp]: ok

**Review by simzer — APPROVED** [review]

**Pull request merged**